reflectx/reflect.go

```diff
@@ -6,20 +6,63 @@
 //
 package reflectx
 
-import "sync"
-
 import (
+	"fmt"
 	"reflect"
 	"runtime"
+	"strings"
+	"sync"
 )
 
-type fieldMap map[string][]int
+type field struct {
+	Index    []int
+	Path     string
+	Field    reflect.StructField
+	Zero     reflect.Value
+	Name     string
+	Options  map[string]string
+	Embedded bool
+}
+
+type fields struct {
+	Index []*field
+	Paths map[string]*field
+	Names map[string]*field
+}
+
+func (f fields) GetByPath(path string) *field {
+	if fi, ok := f.Paths[path]; ok {
+		return fi
+	} else {
+		return nil
+	}
+}
+
+func (f fields) GetByTraversal(index []int) *field {
```

Comment:
Might be easier as `GetByTraversal(indexes ...int)`

Shouldn't this be more or less the same as [FieldByIndex(index []int) Value](http://golang.org/src/reflect/value.go?s=22469:22515#L726)?   The only reason that `FieldByIndexes` is different from `reflect.FieldByIndex` is that it is designed to not panic if it encounters nil pointers to embedded structs, as it's designed to be a bit more convenient for unmarshallers.  That should really be documented somewhere.